ethcore/src/snapshot/service.rs

```diff
@@ -492,6 +580,7 @@ impl Service {
 
 		let _ = fs::remove_dir_all(self.restoration_dir());
 		*self.status.lock() = RestorationStatus::Inactive;
+		self.ready.store(false, Ordering::SeqCst);
```

Comment:
Is that fine? Why we are becoming not ready after the snapshot is restored? Maybe the name is not good enough and should encapsulate more regarding "readiness"

Reply:
Yeah so `ready` here kind of means `initialized`. The issue being that it should be set to `false` before `init_restore` is called, since it's called by message passing and otherwise there would be race condition with the Chain Sync reading the `ready` value before the message is passed and `init_restore` is called.

Reply:
I see. Is it possible to encapsulate that flow in the names somehow? Currently it's a little bit misleading, maybe call it `is_ready_for_restoration`?